codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpProtocolUnitTestRequestGenerator.java

```diff
@@ -250,20 +241,21 @@ protected void generateTestAssertions(GoWriter writer) {
         writeAssertNil(writer, "err");
         writeAssertNotNil(writer, "result");
 
-        writeAssertScalarEqual(writer, "c.ExpectMethod", "actualReq.Method", "method");
-        writeAssertScalarEqual(writer, "c.ExpectURIPath", "actualReq.URL.RawPath", "path");
+        writeAssertScalarEqual(writer, "c.ExpectMethod", "capturedReq.Method", "method");
```

Comment:
This is totally fine but you might have been able to get away with less changes by just connecting your new middleware to the originally declared `actualReq` request.